examples/visionProcessing/faceDetection/faceDetection.cpp

```diff
@@ -0,0 +1,249 @@
+    cv::Size size = cv::Size(300, 300);
+    // cv::Scalar meanVal = cv::Scalar(104, 117, 123);
+    cv::Scalar MODEL_MEAN_VALUES =
+        cv::Scalar(78.4263377603, 87.7689143744, 114.895847746);
+
+    cv::Mat inputBlob = cv::dnn::blobFromImage(
+        frameOpenCVDNN, inScaleFactor, size, MODEL_MEAN_VALUES, true, false);
+
+    net.setInput(inputBlob, "data");
+    cv::Mat detection = net.forward("detection_out");
+
+    cv::Mat detectionMat(detection.size[2], detection.size[3], CV_32F,
+                         detection.ptr<float>());
+
+    std::vector<std::vector<int>> rectangles;
+
+    for (int i = 0; i < detectionMat.rows; i++) {
+      float confidence = detectionMat.at<float>(i, 2);
+
+      if (confidence > confidenceThreshold) {
+        int x1 = static_cast<int>(detectionMat.at<float>(i, 3) * frameWidth);
+        int y1 = static_cast<int>(detectionMat.at<float>(i, 4) * frameHeight);
+        int x2 = static_cast<int>(detectionMat.at<float>(i, 5) * frameWidth);
+        int y2 = static_cast<int>(detectionMat.at<float>(i, 6) * frameHeight);
+        std::vector<int> box = {x1, y1, x2, y2};
+        rectangles.push_back(box);
+        cv::rectangle(frameOpenCVDNN, cv::Point(x1, y1), cv::Point(x2, y2),
+                      cv::Scalar(0, 255, 0), 2, 4);
+      }
+    }
+
+    return std::make_tuple(frameOpenCVDNN, rectangles);
+  }
+
+private:
+  std::vector<std::vector<int>> bboxes;
+  cv::Mat frameFace;
+  int padding = 20;
+  int computeCounter = 0;
+  cv::dnn::Net faceNet;
+};
+
+std::vector<uint32_t> createConsumerPartitions(uint32_t highestPartitionNumber,
+                                               uint32_t consumerIndex,
+                                               uint32_t numberOfConsumers) {
+  std::vector<uint32_t> consumerPartitions;
+  for (auto i = consumerIndex; i <= highestPartitionNumber;
+       i += numberOfConsumers) {
+    consumerPartitions.push_back(i);
+  }
+  return consumerPartitions;
+}
+
+void run(const std::string &nodeName, const std::string &dagFileName,
+         uint32_t consumerIndex, uint32_t numberOfConsumers,
+         const std::string &protobufFile, const std::string &mlModel) {
+
+  FaceDetection facialDetection(protobufFile, mlModel);
+  ndn::Face face;
+
+  auto dagParser = iceflow::DAGParser::parseFromFile(dagFileName);
+  auto node = dagParser.findNodeByName(nodeName);
+  if (!node.downstream.empty()) {
+    std::string downstreamEdgeNameAD, downstreamEdgeNameGD;
+    for (const auto &downstream : node.downstream) {
+      if (downstream.id == "fd2ad") {
+        downstreamEdgeNameAD = downstream.id;
+      } else if (downstream.id == "fd2gd") {
+        downstreamEdgeNameGD = downstream.id;
+      }
+    }
+
+    if (downstreamEdgeNameAD.empty() || downstreamEdgeNameGD.empty()) {
+      std::cerr << "Error: Missing downstream targets for face detection or "
+                   "people counting."
+                << std::endl;
+      return;
+    }
```

Comment:
I think the logic here could be simplified by using something like `std::find`. And even then, since you have hardcoded the edge names anyway, you could also consider using them right away.